src/jit/compiler.cpp

```diff
@@ -2440,147 +2484,111 @@ void Compiler::compSetProcessor()
 
     if (!jitFlags.IsSet(JitFlags::JIT_FLAG_PREJIT))
     {
-        if (configEnableISA(InstructionSet_Base))
+        if (configEnableIsa(jitFlags, InstructionSet_Base))
         {
             opts.setSupportedISA(InstructionSet_Base);
-        }
-        if (configEnableISA(InstructionSet_SSE))
-        {
-            opts.setSupportedISA(InstructionSet_SSE);
+
+            if (configEnableIsa(jitFlags, InstructionSet_SSE))
+            {
+                opts.setSupportedISA(InstructionSet_SSE);
 #ifdef _TARGET_AMD64_
-            opts.setSupportedISA(InstructionSet_SSE_X64);
-#endif
-        }
-        if (configEnableISA(InstructionSet_SSE2))
-        {
-            opts.setSupportedISA(InstructionSet_SSE2);
+                opts.setSupportedISA(InstructionSet_SSE_X64);
+#endif // _TARGET_AMD64_
+
+                if (configEnableIsa(jitFlags, InstructionSet_SSE2))
+                {
+                    opts.setSupportedISA(InstructionSet_SSE2);
 #ifdef _TARGET_AMD64_
-            opts.setSupportedISA(InstructionSet_SSE2_X64);
-#endif
-        }
-        if (jitFlags.IsSet(JitFlags::JIT_FLAG_USE_LZCNT))
-        {
-            if (configEnableISA(InstructionSet_LZCNT))
-            {
-                opts.setSupportedISA(InstructionSet_LZCNT);
+                    opts.setSupportedISA(InstructionSet_SSE2_X64);
+#endif // _TARGET_AMD64_
+
+                    if (configEnableIsa(jitFlags, InstructionSet_AES))
+                    {
+                        opts.setSupportedISA(InstructionSet_AES);
+                    }
+
+                    if (configEnableIsa(jitFlags, InstructionSet_PCLMULQDQ))
+                    {
+                        opts.setSupportedISA(InstructionSet_PCLMULQDQ);
+                    }
+
+                    // We need to additionaly check that COMPlus_EnableSSE3_4 is set, as that
+                    // is a prexisting config flag that controls the SSE3+ ISAs
+                    if (configEnableIsa(jitFlags, InstructionSet_SSE3) && JitConfig.EnableSSE3_4())
```

Comment:
Given that we have the ability to control each ISA individually, and the knobs work according to the appropriate hierarchy, it would be nice if we could get rid of `EnableSSE3_4` and tell users to change it to `EnableSSE3`.

@CarolEidt, thoughts? I know we had discussed similar changes in the past, and you weren't much in favor of removing or breaking pre-existing configuration knobs. However, this one in particular only impacts the 3 or 4 `S.N.Vector` code paths that use `SIMD_SSE4_Supported` and (at least on 2.2 and prior) only if `AVX` is also disabled.

Reply:
That may make sense. I believe that we introduced the support of the various `COMPlus` knobs for the individual ISAs post-2.2, is that correct? If so, I think it would be mostly an internal impact.

Reply:
> I believe that we introduced the support of the various COMPlus knobs for the individual ISAs post-2.2, is that correct

Correct

> If so, I think it would be mostly an internal impact.

`COMPlus_EnableSSE3_4` in particular came around at the same time as `COMPlus_EnableAVX`, as part of the initial `S.N.Vector` work; and has been a "retail" flag for the same length of time; it just has much less scope and impact than `COMPlus_EnableAVX`